lib/qa/authorities/linked_data/find_term.rb

```diff
@@ -79,82 +80,57 @@ def expects_uri?
           term_config.term_id_expects_uri?
         end
 
-        def extract_uri_from_id(id)
-          return @uri = id if expects_uri?
+        def uri
+          return @uri if @uri.present?
+          return @uri = RDF::URI.new(id) if expects_uri?
           @uri = graph_service.subjects_for_object_value(graph: @filtered_graph, predicate: RDF::URI.new(term_config.term_results_id_predicate), object_value: id.gsub('%20', ' ')).first
         end
 
         def preds_for_term
-          { required: required_term_preds, optional: optional_term_preds }
-        end
-
-        def required_term_preds
           label_pred_uri = term_config.term_results_label_predicate
           raise Qa::InvalidConfiguration, "required label_predicate is missing in configuration for LOD authority #{authority_name}" if label_pred_uri.nil?
-          { label: label_pred_uri }
+          preds = { label: label_pred_uri }
+          preds.merge(optional_preds)
         end
 
-        def optional_term_preds
-          preds = {}
-          preds[:altlabel] = term_config.term_results_altlabel_predicate unless term_config.term_results_altlabel_predicate.nil?
-          preds[:id] = term_config.term_results_id_predicate unless term_config.term_results_id_predicate.nil?
-          preds[:narrower] = term_config.term_results_narrower_predicate unless term_config.term_results_narrower_predicate.nil?
-          preds[:broader] = term_config.term_results_broader_predicate unless term_config.term_results_broader_predicate.nil?
-          preds[:sameas] = term_config.term_results_sameas_predicate unless term_config.term_results_sameas_predicate.nil?
-          preds
+        def optional_preds
+          opt_preds = {}
+          opt_preds[:altlabel] = term_config.term_results_altlabel_predicate
+          opt_preds[:id] = term_config.term_results_id_predicate
+          opt_preds[:narrower] = term_config.term_results_narrower_predicate
+          opt_preds[:broader] = term_config.term_results_broader_predicate
+          opt_preds[:sameas] = term_config.term_results_sameas_predicate
+          opt_preds.delete_if { |_k, v| v.blank? }
```

Comment:
Instead of testing on every line setting optional predicates, simply remove any that are blank.